### PR TITLE
[RESTEASY-1890] [RESTEASY-1892]

### PR DESCRIPTION
**Pull request by liweinan**

### CODE REVIEW

**Comment by liweinan** [comment]
work in progress, don't merge yet.

**Comment by liweinan** [comment]
- [RESTEASY-1890] Implement all the server side tracing events
    - [RESTEASY-1892] Implement the other common tracing events
    - many tracing info errors are fixed
    - tests are updated

**Comment by liweinan** [comment]
Sample tracing info from client side:

image

**Comment by liweinan** [comment]
@asoldano Thanks for checking! Let me fix them and rebase the PR .

File: resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/ResteasyClientBuilder.java [hunk 1]
```diff
@@ -7,7 +7,9 @@
 import org.jboss.resteasy.client.jaxrs.internal.ClientConfiguration;
 import org.jboss.resteasy.client.jaxrs.internal.LocalResteasyProviderFactory;
 import org.jboss.resteasy.plugins.providers.RegisterBuiltin;
+import org.jboss.resteasy.plugins.server.servlet.ResteasyContextParameters;
```
[the remark this answers is not in the document]
Comment (liweinan) [reply resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/ResteasyClientBuilder.java]: This is unnecessary, Let me clean up it.

File: resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/ResteasyClientBuilder.java [hunk 2]
```diff
@@ -7,7 +7,9 @@
 import org.jboss.resteasy.client.jaxrs.internal.ClientConfiguration;
 import org.jboss.resteasy.client.jaxrs.internal.LocalResteasyProviderFactory;
 import org.jboss.resteasy.plugins.providers.RegisterBuiltin;
+import org.jboss.resteasy.plugins.server.servlet.ResteasyContextParameters;
 import org.jboss.resteasy.spi.ResteasyProviderFactory;
+import org.jboss.resteasy.tracing.RESTEasyTracingUtils;
```
[the remark this answers is not in the document]
Comment (liweinan) [reply resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/ResteasyClientBuilder.java]: This is unnecessary, Let me clean up it.

File: resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/interception/jaxrs/ServerReaderInterceptorContext.java [hunk 3]
```diff
@@ -33,7 +34,8 @@ public ServerReaderInterceptorContext(ReaderInterceptor[] interceptors, Resteasy
                                          MultivaluedMap<String, String> headers, InputStream inputStream,
                                          HttpRequest request)
    {
-      super(mediaType, providerFactory, annotations, interceptors, headers, genericType, type, inputStream);
+//      super(mediaType, providerFactory, annotations, interceptors, headers, genericType, type, inputStream);
```
[the remark this answers is not in the document]
Comment (liweinan) [reply resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/interception/jaxrs/ServerReaderInterceptorContext.java]: done.

File: resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/interception/jaxrs/ServerWriterInterceptorContext.java [hunk 4]
```diff
@@ -30,7 +32,9 @@ public ServerWriterInterceptorContext(WriterInterceptor[] interceptors, Resteasy
                                          OutputStream outputStream,
                                          HttpRequest request)
    {
-      super(interceptors, annotations, entity, genericType, mediaType, type, outputStream, providerFactory, headers);
+//      super(interceptors, annotations, entity, genericType, mediaType, type, outputStream, providerFactory, headers);
```
[the remark this answers is not in the document]
Comment (liweinan) [reply resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/interception/jaxrs/ServerWriterInterceptorContext.java]: done.

File: resteasy-jaxrs/src/main/java/org/jboss/resteasy/specimpl/BuiltResponse.java [hunk 5]
```diff
@@ -6,6 +6,7 @@
 import org.jboss.resteasy.spi.HeaderValueProcessor;
 import org.jboss.resteasy.spi.LinkHeaders;
 import org.jboss.resteasy.spi.ResteasyProviderFactory;
+import org.jboss.resteasy.tracing.RESTEasyTracingLogger;
```
[the remark this answers is not in the document]
Comment (liweinan) [reply resteasy-jaxrs/src/main/java/org/jboss/resteasy/specimpl/BuiltResponse.java]: it's unnecessary, let me remove it.

File: resteasy-jaxrs/src/main/java/org/jboss/resteasy/spi/ResteasyProviderFactory.java [hunk 6]
```diff
@@ -2708,7 +2797,8 @@ public void injectProperties(Object obj, HttpRequest request, HttpResponse respo
 
 
    // Configurable
-
+   // Please use `propery()` function because it's standard API.
+   @Deprecated
    public Map<String, Object> getMutableProperties()
```
[the remark this answers is not in the document]
Comment (liweinan) [reply resteasy-jaxrs/src/main/java/org/jboss/resteasy/spi/ResteasyProviderFactory.java]: okay, let me work on this in a separate PR(and create a JIRA issue for it).

File: tjws/src/main/java/org/jboss/resteasy/test/BaseResourceTest.java [hunk 7]
```diff
@@ -9,6 +9,7 @@
 import org.jboss.resteasy.spi.ResteasyProviderFactory;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
+import org.junit.Ignore;
```
[the remark this answers is not in the document]
Comment (liweinan) [reply tjws/src/main/java/org/jboss/resteasy/test/BaseResourceTest.java]: it's not needed, let me remove it.

**Comment by liweinan** [comment]
rebase done, next step is to migrate tests.

File: testsuite/integration-tests/pom.xml [hunk 8]
```diff
@@ -74,6 +74,13 @@
                                        dest="${project.build.directory}/test-server/wildfly-${server.version}/modules/system/layers/base"
                                        overwrite="true"/>
                                 <delete dir="target/dependency-maven-plugin-markers"/>
+                                <replace file="${project.build.directory}/test-server/wildfly-${server.version}/standalone/configuration/standalone-full.xml">
+                                    <replacetoken><![CDATA[="INFO"]]></replacetoken>
+                                    <replacevalue><![CDATA[="DEBUG"]]></replacevalue>
```
Comment (liweinan) [inline testsuite/integration-tests/pom.xml]: We don't need a separate container for tracing test, we just need to switch the log output to DEBUG for tracing test. @asoldano If you agree, I'll keep the solution like this.

**Comment by liweinan** [comment]
surefire execution removed for tracing test.

File: testsuite/integration-tests/pom.xml (hunk 8 above)
Comment (asoldano) [reply testsuite/integration-tests/pom.xml]: @liweinan does that mean that the whole testsuite will run with the server having debug level logging enabled? If that's the case, I'd rather avoid this and have a separate container instance for the tracing tests.

Comment (liweinan) [reply testsuite/integration-tests/pom.xml]: @asoldano okay then let me create a separate container for this.

**Comment by liweinan** [comment]
After discussing with @asoldano , the solution is like this:

- Make a new copy of `standalone.xml` with replaced logging level
- Let the tracing container use the customized configuration file.

File: testsuite/integration-tests/src/test/resources/arquillian.xml [hunk 9]
```diff
@@ -26,6 +26,17 @@
                 <property name="managementPort">10990</property><!-- keep in sync with port-offset -->
             </configuration>
         </container>
+        <container qualifier="jbossas-manual-tracing" mode="manual">
+            <configuration>
+                <property name="jbossHome">${jboss.home}</property>
+                <property name="jbossArguments">${securityManagerArg}</property>
+                <property name="serverConfig">standalone-tracing.xml</property>
```
Comment (liweinan) [inline testsuite/integration-tests/src/test/resources/arquillian.xml]: the tracing container will use separate configuration.

File: testsuite/integration-tests/pom.xml [hunk 10]
```diff
@@ -74,6 +74,13 @@
                                        dest="${project.build.directory}/test-server/wildfly-${server.version}/modules/system/layers/base"
                                        overwrite="true"/>
                                 <delete dir="target/dependency-maven-plugin-markers"/>
+                                <copy file="${project.build.directory}/test-server/wildfly-${server.version}/standalone/configuration/standalone-full.xml"
```
Comment (liweinan) [inline testsuite/integration-tests/pom.xml]: A copy of configuration for tracing testing.

**Pull request merged**